packages/grid/_modules_/grid/models/api/gridEditingApi.ts

```diff
@@ -69,10 +82,22 @@ export interface GridEditRowApi {
     event?: MuiBaseEvent,
   ) => boolean | Promise<boolean>;
   /**
-   * Updates the row at the given id with the values stored in the edit row model.
-   * @param {GridRowId} id The id to commit to.
-   * @param {React.SyntheticEvent} event The event to pass forward.
-   * @returns {boolean} A boolean indicating if there is an error.
+   * Sets the mode of a cell.
+   * @param {GridRowId} id The id of the row.
+   * @param {string} field The field to change the mode.
+   * @param {GridCellMode} mode Can be: `"edit"`, `"view"`.
    */
-  commitRowChange: (id: GridRowId, event?: MuiBaseEvent) => boolean | Promise<boolean>;
+  setCellMode: (id: GridRowId, field: string, mode: GridCellMode) => void;
+  /**
+   * Gets the mode of a cell.
+   * @param {GridRowId} id The id of the row.
+   * @param {string} field The field to get the mode.
+   * @returns {GridCellMode} Returns `"edit"` or `"view"`.
+   */
+  getCellMode: (id: GridRowId, field: string) => GridCellMode;
 }
+
+/**
+ * The editing API interface that is available in the grid `apiRef`.
+ */
+export interface GridEditingApi extends GridCellEditingApi, GridRowEditingApi {}
```

Comment:
Just a slight follow up on the typing breaking change
This is technically a typing breaking change for users importing the old interface.
We can keep exporting a deprecated alias `export type GridEditingApi = GridEditingApi` or consider that this is a very minor BC and do nothing more than your current PR.